src/os/data/registry.js

```diff
@@ -182,6 +182,15 @@ class Registry extends EventTarget {
     }
     return false;
   }
+
+  /**
+   * Empty out the Registry
+   */
+  clear() {
+    this.keys().forEach((key) => {
```

Comment:
I was worried about places that currently use Registries will not have listeners set up for a new event -- so as you said I was piggy-backing on the "remove" event handlers.

However, it makes sense to have a "clear" event be part of the contract for the case of large registries. Might even be worth making the "clear" event pass an array of registry entries so it wouldn't be hard to later upgrade the single-value handlers to handle batches if they want to support the clear event.

Reply:
Updated. For the on() helper, I just hooked back into the passed-in onRemove handler for the clear() -- it's not great, but it gives backward compatibility to the few places using the helper method.
It might be worth doing some refactoring during the OS update. TBD.